app/models/solid_cache/entry.rb

```diff
@@ -140,14 +140,25 @@ def to_binary(key)
 
         def expiry_candidate_ids(count, max_age:, max_entries:)
           cache_full = max_entries && max_entries < id_range
-          min_created_at = max_age.seconds.ago
 
-          uncached do
-            order(:id)
-              .limit(count * 3)
-              .pluck(:id, :created_at)
-              .filter_map { |id, created_at| id if cache_full || created_at < min_created_at }
-              .sample(count)
+          if max_age
+            min_created_at = max_age.seconds.ago
+
+            uncached do
+              order(:id)
+                .limit(count * 3)
+                .pluck(:id, :created_at)
+                .filter_map { |id, created_at| id if cache_full || created_at < min_created_at }
+                .sample(count)
+            end
+          elsif cache_full
+            uncached do
+              order(:id)
+                .limit(count)
```

Comment:
Could you do `.limit(count * 3)` and `.sample(count)` here like in the other query?

The idea is when you have concurrent attempts to delete records, by choosing a random selection of the latest candidates, you reduce the overlap between them. 

Whether that's a useful thing to do in practice is maybe debatable and this might be removed, but we should be consistent until then.

Reply:
Ah, I was assuming the extra records were to compensate for reducing the count with `filter_map`. I restored that behavior and added a comment and local variable to show the intent.